onionshare/web/chat_mode.py

```diff
@@ -80,40 +91,33 @@ def update_username(message):
             The message is sent to all people in the room."""
             current_name = session.get("name")
             session["name"] = message["username"]
-            session["worker"].stop_thread()
-            session["worker"] = UserListWorker(self.web.socketio)
-            session['thread'] = self.web.socketio.start_background_task(
-                session["worker"].background_thread, session['name']
-            )
+            self.connected_users[
+                self.connected_users.index(current_name)
+            ] = session.get("name")
             emit(
                 "status",
-                {"msg": current_name + " has updated their username to: " + session.get("name"),
-                "old_name": current_name,
-                "new_name": session.get("name")
+                {
+                    "msg": "{} has updated their username to: {}".format(
+                        current_name, session.get("name")
+                    ),
+                    "connected_users": self.connected_users,
+                    "old_name": current_name,
+                    "new_name": session.get("name"),
                 },
-                room=session.get("room")
+                room=session.get("room"),
             )
 
-
-
-class UserListWorker(object):
```

Comment:
Removed the background worker logic by an instance variable logic. Given that we can assume that onionshare chat will be running on a single server, and not clustered over different servers, the instance variable will solve the purpose of holding an user list (Point me if I am wrong). This would reduce the load because no background thread.